src/main/java/longah/util/GroupList.java

```diff
@@ -0,0 +1,224 @@
+ * Represents a list of groups in the LongAh application.
+ */
+public class GroupList {
+    private static final String GROUP_LIST_FILE_PATH = "./data/groupList.txt";
+    private static Group activeGroup;
+    private static ArrayList<Group> groupList = new ArrayList<>();
+
+    /**
+     * Constructor for GroupList.
+     *
+     * @throws LongAhException If an I/O exception occurs.
+     */
+    public GroupList() throws LongAhException {
+        StorageHandler.initDir();
+        if (!Files.exists(Paths.get(GROUP_LIST_FILE_PATH)) || groupList == null) {
+            createGroupList();
+        } else {
+            loadGroupList();
+            getGroupList();
+            UI.showMessage("Defaulting to the first group. You are now managing: "
+                    + groupList.get(0).getGroupName());
+            activeGroup = groupList.get(0);
+        }
+    }
+
+    /**
+     * Returns the active group.
+     *
+     * @return The active group.
+     */
+    public static Group getActiveGroup() {
+        return activeGroup;
+    }
+
+    /**
+     * Switches the active group to the specified group.
+     *
+     * @param newGroup The new group to switch to.
+     */
+    public static void switchActiveGroup(Group newGroup) {
+        activeGroup = newGroup;
+    }
+
+    /**
+     * Creates a new group list.
+     * @throws LongAhException If an I/O exception occurs.
+     */
+    public static void createGroupList() throws LongAhException {
+        groupList = new ArrayList<>();
+        UI.showMessage("No groups found. Please give a name for your first group.");
+        String groupName = UI.getUserInput();
+        Group newGroup = new Group(groupName);
+        groupList.add(newGroup);
+        try {
+            Files.write(Paths.get(GROUP_LIST_FILE_PATH), groupName.getBytes());
+        } catch (IOException e) {
+            throw new LongAhException(ExceptionMessage.IO_EXCEPTION);
+        }
+        activeGroup = newGroup;
+    }
+
+    /**
+     * Loads the group list from the file.
+     * @throws LongAhException If an I/O exception occurs.
+     */
+    public static void loadGroupList() throws LongAhException {
+        try {
+            String[] data = new String(Files.readAllBytes(Paths.get(GROUP_LIST_FILE_PATH))).split("\n");
+            if (data.length == 0) {
+                throw new LongAhException(ExceptionMessage.EMPTY_GROUP_LIST);
+            }
+            for (String groupName : data) {
+                Group newGroup = new Group(groupName);
+                groupList.add(newGroup);
+            }
+        } catch (IOException e) {
+            throw new LongAhException(ExceptionMessage.IO_EXCEPTION);
+        }
+    }
+
+    /**
+     * Returns the group list.
+     *
+     * @return The group list.
+     * @throws LongAhException If an I/O exception occurs.
+     */
+    public static String getGroupList() throws LongAhException {
+        // did not use exceptions here as I want to return an empty string
```

Comment:
If list is empty it will throw the IO exception though, might want to add a return statement after showing empty list message.